### PR TITLE
Fix light requests handler error handling and logging target

### PR DESCRIPTION
**Pull request by tomaka**
Adds `target: LOG_TARGET` everywhere it is missing, and properly returns an errors instead of returning empty successful results.

### CODE REVIEW

**Review by expenses — APPROVED** [review]
Fantastic! 👍🏻

File: client/network/src/service.rs [hunk 1]
```diff
@@ -1528,6 +1528,7 @@ impl<B: BlockT + 'static, H: ExHashT> Future for NetworkWorker<B, H> {
 					}
 					{
 						let mut peers_notifications_sinks = this.peers_notifications_sinks.lock();
+						println!("peers_notifications_sinks.insert({:?}, {:?})", remote, protocol);
```
Comment (bkchr) [inline client/network/src/service.rs:1531]: Pierre :D

Comment (tomaka) [reply client/network/src/service.rs:1531]: Ah fuck

**Review by mxinden — APPROVED** [review]

**Comment by tomaka** [comment]
I have absolutely no idea how to fix these tests.

**Comment by expenses** [comment]
> I have absolutely no idea how to fix these tests.

Merging with master fixed it for me

**Review by romanb — APPROVED** [review]

**Comment by gnunicorn** [comment]
so... is there anyone working on fixing these tests?

**Comment by tomaka** [comment]
That should be me.
The problem I face is that the tests strongly assume that a request is always successful, which is something this PR modifies. I would need to refactor the tests, but they're very weirdly written.

**Comment by gnunicorn** [comment]
closing due to a lack of progress.

**Pull request closed without merging**